doc/userguide/devguide/codebase/contributing/contribution-process.rst

```diff
@@ -0,0 +1,135 @@
+************************
+Contributing to Suricata
+************************
+
+This guide describes what steps to take if you want to contribute a patch or
+patchset to Suricata.
+Before you start, please review and sign our `Contribution Agreement
+<https://suricata.io/contribution-agreements/>`_.
+
+Open (or claim) a ticket
+========================
+
+For features and bugs we need `tickets <https://redmine.openinfosecfoundation.org/projects/suricata/issues>`_. Tickets
+help us keep track of the work done, indicate when changes need backports etc.
+
+They are also important if you would like to see your new feature officially
+added to our tool: the ticket documents your ideas so  we can analyze how do they
+fit in our plans for Suricata, and, if the feature is accepted, we can properly
+track progress etc...
+
+For really trivial fixes or cleanups we won't need that.
+
+Assign the ticket to yourself. For this you will need to have the "developer"
+role. Please get in touch with us to request this role.
+
+If a ticket is already assigned to someone, please reach out on the ticket or 
+ask the person first.
+You can reach out to other community members via the `Suricata Discord server
+<https://discord.com/invite/t3rV2x7MrG>`_.
+
+Expectations
+-------------
+
+If you submit a new feature that is not part of Suricata's core
+functionalities, we would expect some commitment from you or the organization
+who is sponsoring your work, before we could approve the new feature, as the
+Suricata development team is pretty lean (and many times overworked).
+
+This means we expect that
+
+    * the new contribution comes with a set of Suricata-verify tests (and
+      possibly unittests, where those apply), before we can approve it;
+    * proof of compability with existing keywords/features is provided,
+      when the contribution is for replacing an existing feature;
+    * you would maintain the feature once it is approved - or find someone who
+      can do that, in case you cannot;
+    * you respect our guidelines and processes. We appreciate community
+      contributors: Suricata wouldn't be what it is without them; and the
+      value of our tool and community also comes from how seriously we take all
+      this, and we ask that our contributors do the same! ;)
+
+
+What branch to work on
+======================
+
+There are 2 active branches:
```

Comment:
2 or 3, but ultimately its about new dev being for master only except in well exceptional cases :)

Reply:
roger that!